website/docs/reference/resource-properties/constraints.md

```diff
@@ -3,21 +3,22 @@ resource_types: [models]
 datatype: "{dictionary}"
 ---
 
-:::caution Under construction 🚧
-These docs are liable to change!
-:::
-
 :::info Beta functionality
-This functionality is new in v1.5! These docs exist to provide a high-level overview of what's to come. Specific syntax is liable to change.
-
-For more details, and to leave your feedback, check out this GitHub issue:
-* ["Unify constraints and constraints_check configs" (dbt-core#6750)](https://github.com/dbt-labs/dbt-core/issues/6750)
+This functionality is new in v1.5! The syntax is mostly locked, but some small details are still liable to change.
 :::
 
-In transactional databases, it is possible to define "constraints" on the allowed values of certain columns, stricter than just the data type of those values. Because Postgres is a transactional database, it supports and enforces all the constraints in the ANSI SQL standard (`not null`, `unique`, `primary key`, `foreign key`), plus a flexible row-level `check` constraint that evaluates to a boolean expression.
+The structure of a constraint is:
+- `type`: one of `not_null`, `primary_key`, `foreign_key`, `check`, `custom`
+- `expression`: text input to qualify the constraint
+
+In transactional databases, it is possible to define "constraints" on the allowed values of certain columns, stricter than just the data type of those values. For example, Postgres supports and enforces all the constraints in the ANSI SQL standard (`not null`, `unique`, `primary key`, `foreign key`), plus a flexible row-level `check` constraint that evaluates to a boolean expression.
 
 Most analytical data platforms support and enforce a `not null` constraint, but they either do not support or do not enforce the rest. It is sometimes still desirable to add an "informational" constraint, knowing it is _not_ enforced, for the purpose of integrating with legacy data catalog or entity-relation diagram tools ([dbt-core#3295](https://github.com/dbt-labs/dbt-core/issues/3295)).
 
+To that end, there are two optional fields you can specify on any constraint:
+- `warn_unenforced: False` to skip warning on constraints that are supported, but not enforced, by this data platform. The constraint will be included in templated DDL.
+- `warn_unsupportedL: False` to skip warning on constraints that aren't supported by this data platform, and therefore won't be included in templated DDL.
+
```

Comment:
TODO: document the limitations called out in https://github.com/dbt-labs/docs.getdbt.com/issues/2945#issuecomment-1468821858

> - no `ref` support for `foreign_key` / `references` constraints
> - no standard support (yet) for platform-specific `rely` instructions, unless implemented for that platform specifically
> - no support for schema evolution in `incremental` models - adding/removing constraints as the column spec changes, during incremental runs